File: kt/BeginnersSelection/06_CardGameForTwo.kt
// https://atcoder.jp/contests/abs/tasks/abc088_b

fun main(args: Array<String>) = (
  readLine()!!
    .let { _ ->
      readLine()!!
        .split(" ")
        .map(String::toInt)
        .sorted()
        .reversed()
        .withIndex()
        .fold(0 to 0, { a, x ->
          if (x.index % 2 == 0)
            a.first + x.value to a.second
          else
            a.first to a.second + x.value
        })
    }
    .let { (a, b) -> a - b }
    .let(::println)
)

// https://atcoder.jp/contests/abs/submissions/21021940

File: jl/BeginnersSelection/06_CardGameForTwo.jl
# https://atcoder.jp/contests/abs/tasks/abc088_b

(
  (readline(), readline())[2]
  |> split
  |> a -> parse.(Int, a)
  |> a -> sort(a, rev = true)
  |> a -> foldl(((a, b, i), x) -> (
    i % 2 == 0
      ? (a + x, b, i + 1)
      : (a, b + x, i + 1)
  ), a, init = (0, 0, 0))
  |> x -> x[1] - x[2]
  |> println
)

# https://atcoder.jp/contests/abs/submissions/22656992
